app/Http/Controllers/ControllerDash.php

```diff
@@ -211,19 +213,31 @@ public function showProfile($year = null, $month = null) {
             $last_training_given = null;
         }
 
-        $setmore_appointments = SetmoreAppointment::where('customer_cid', Auth::id())->get();
-        foreach ($setmore_appointments as &$setmore_appointment) {
-            $setmore_appointment->res_date = Carbon::parse($setmore_appointment->start_time)->format('m/d/Y');
-            $setmore_appointment->res_time = Carbon::parse($setmore_appointment->start_time)->format('H:i');
+        if (is_null(Auth::user()->ea_customer_id)) {
+            $ea_users = DB::connection('ea_mysql')->table('ea_users')->select('id')->where('email', Auth::user()->email)->where('id_roles', '3')->limit(1)->get();
+            foreach ($ea_users as $u) {
+                $user = Auth::user();
+                $user->ea_customer_id = $u->id;
+                $user->save();
+            }
         }
-
-        $setmore_data_stale = false;
-        $setmore_recent_pulls = SetmoreAppointment::where('created_at', '>', Carbon::now()->subHours(1)->toDateTimeString())->count();
-        if ($setmore_recent_pulls == 0) {
-            $setmore_data_stale = true;
+        $ea_appointments = [];
```

Comment:
This doesn't seem to detect cancellations... not entirely sure why. Perhaps it doesn't remove it from the table?

Reply:
It does detect reschedules and changes successfully though

Reply:
Fixed

Reply:
Looks great!